### PR TITLE
Documentation for Urgent Message and HostedContent support

### PR DESCRIPTION
**Pull request by RamjotSingh**
Adding documentation for support of urgent message. Also updating some outdated documentation

### CODE REVIEW

**Comment by RamjotSingh** [comment]
@nkramer Please review.

File: api-reference/beta/resources/chatMessageType.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Represents type of an individual [chat message](chatMessage.md) within a [channel](channel.md) or [chat](chat.md) entity.
+
+## Members
+
+| Member    | Value | Description                               | Supported     |
+| :-------- | :---- | :---------------------------------------- | :------------ |
+| message   | 0     | Message type as a text or an HTML message | Supported     |
+| chatEvent | 1     | Conversation event                        | Not supported |
```
Comment (nkramer) [inline api-reference/beta/resources/chatMessageType.md]: What's the point of documenting the values we don't support?

Comment (RamjotSingh) [reply api-reference/beta/resources/chatMessageType.md]: I can pull them out.

**Comment by RamjotSingh** [comment]
@Lauragra Can you merge this PR?

File: api-reference/beta/resources/chatMessageImportance.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+---
```
Comment (Lauragra) [inline api-reference/beta/resources/chatMessageImportance.md]: Please change the file name to use all lowercase letters.

File: api-reference/beta/resources/chatMessageImportance.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+title: "chatMessageImportance resource type"
+description: "Represents the importance of an individual chat message within a channel or chat entity."
+localization_priority: Normal
+author: "RamjotSingh"
+ms.prod: "microsoft-teams"
+---
+
+# chatMessageImportance resource type
```
Comment (Lauragra) [inline api-reference/beta/resources/chatMessageImportance.md]: This topic does not reflect the standard structure for a resource topic. Is this an enumeration? Please see the guidance here: https://msgo.azurewebsites.net/add/document/guidelines/enumerations.html. Also, did you use the stub generator tool to generate your topics? That ensures that the content conforms to standards. More details here: https://msgo.azurewebsites.net/add/document/guidelines/stub-generator.html.

File: api-reference/beta/resources/chatMessageImportance.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+---
+title: "chatMessageImportance resource type"
+description: "Represents the importance of an individual chat message within a channel or chat entity."
+localization_priority: Normal
+author: "RamjotSingh"
+ms.prod: "microsoft-teams"
```
Comment (Lauragra) [inline api-reference/beta/resources/chatMessageImportance.md]: Global: The topics are missing the doc_type attribute. Please add with the correct values. Details in this topic: https://msgo.azurewebsites.net/add/document/guidelines/metadata.html.

File: api-reference/beta/resources/chatMessageType.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+---
```
Comment (Lauragra) [inline api-reference/beta/resources/chatMessageType.md]: Same comments as per previous topic.

File: concepts/changelog.md [hunk 6]
```diff
@@ -13,6 +13,14 @@ For details about known issues with Microsoft Graph APIs, see [Known issues](kno
 
 ## May 2019
 
+### Microsoft Teams APIs
+
+| **Change type** | **Version** | **Description**                  |
+|:----------------|:------------|:-----------------------------------------|
+| Addition        | Beta        | Added **urgent** property to the [chatMessageImportance](/graph/api/resources/chatMessageImportance?view=graph-rest-beta) enum.|
+| Addition        | Beta        | Added **hostedContents** navigation property to the [chatMessage](/graph/api/resources/chatmessage?view=graph-rest-beta) entity.|
```
Comment (Lauragra) [inline concepts/changelog.md]: I don't see this update to the chatMessage topic.

**Comment by Lauragra** [comment]
@RamjotSingh , this PR is failing validation. Please review the validation errors and update the content.

File: api-reference/beta/resources/chatMessageImportance.md (hunk 4 above)
Comment (RamjotSingh) [reply api-reference/beta/resources/chatMessageImportance.md]: Used stub generator to update the files.

File: api-reference/beta/resources/chatMessageImportance.md (hunk 3 above)
Comment (RamjotSingh) [reply api-reference/beta/resources/chatMessageImportance.md]: Done. Used stub generator to update files.

**Comment by RamjotSingh** [comment]
@nkramer Please review.
@Lauragra After running the Stub generator. It asked me to rename a file which resulted into TOC getting touched. I understand the documentation says that I should contact docs v-team for such updates. So please do an extra passthrough.

File: api-reference/beta/api/channel-post-messages.md [hunk 7]
```diff
@@ -1,76 +1,86 @@
...
+doc_type: "apiPageType"
 ---
 
-# Send a message to a channel
+# Create chatMessage
 
 [!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
 
-Create a new [message](../resources/chatmessage.md) in the specified [channel](../resources/channel.md).
+Use this API to create a new chatMessage.
```
Comment (Lauragra) [inline api-reference/beta/api/channel-post-messages.md]: Please restore the description that was deleted/replaced. The standard is to begin the summary with an imperative verb (Create) and to link to the associated resource(s).

File: api-reference/beta/api/channel-post-messages.md [hunk 8]
```diff
@@ -1,76 +1,86 @@
...
 ## Response
 
-If successful, this method returns `201 Created` response code with the [message](../resources/chatmessage.md) that was created.
+If successful, this method returns `201, Created` response code and a new [chatMessage](../resources/chatmessage.md) object in the response body.
 
-## Examples 
+## Examples
 
-### Example 1: Hello World
+### Request
```
Comment (Lauragra) [inline api-reference/beta/api/channel-post-messages.md]: Please restore the original H3 here. If you have more than one example in the Examples section, you add H3s for each, with headings Example 1: {description}, Example 2: {description}, and so on.

File: api-reference/beta/api/chat-list-members.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+author: "RamjotSingh"
+ms.prod: "microsoft-teams"
+doc_type: "apiPageType"
+---
+
+# List members
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Retrieve a list of conversationmember objects.
```
Comment (Lauragra) [inline api-reference/beta/api/chat-list-members.md]: Use the correct casing for the resource name (conversationMember) and add a link. Pls. apply fix to the description as well.

File: api-reference/beta/api/chat-list-members.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+| Delegated (work or school account)     | Chat.Read, Chat.ReadWrite |
+| Delegated (personal Microsoft account) | Not supported. |
+| Application                            | Not supported. |
+
+## HTTP request
+
+<!-- { "blockType": "ignored" } -->
+
+```http
+GET /chats/{id}/members
```
Comment (Lauragra) [inline api-reference/beta/api/chat-list-members.md]: Is it members or conversationMembers?

Comment (RamjotSingh) [reply api-reference/beta/api/chat-list-members.md]: It is members.

File: api-reference/beta/api/chat-post-messages.md [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+author: "RamjotSingh"
+ms.prod: "microsoft-teams"
+doc_type: "apiPageType"
+---
+
+# Create chatMessage
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Use this API to create a new chatMessage.
```
Comment (Lauragra) [inline api-reference/beta/api/chat-post-messages.md]: Revise as "Create a new chatMessage" and make the resource name a link.

File: api-reference/beta/api/chat-post-messages.md [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+|:--------------|:--------------|
+| Authorization | Bearer {code} |
+
+## Request body
+
+In the request body, supply a JSON representation of [chatMessage](../resources/chatmessage.md) object.
+
+## Response
+
+If successful, this method returns `201, Created` response code and a new [chatMessage](../resources/chatmessage.md) object in the response body.
```
Comment (Lauragra) [inline api-reference/beta/api/chat-post-messages.md]: Add the word "a" before the HTTPS code and remove the comma after 201.

File: api-reference/beta/api/chat-post-messages.md (hunk 11 above)
Comment (RamjotSingh) [reply api-reference/beta/api/chat-post-messages.md]: Bringing it in sync with the same operation under channel.

**Comment by RamjotSingh** [comment]
@Lauragra I resolved all the comments. Please take another look when you get a chance.

File: api-reference/beta/api/chatmessage-get.md [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+author: "RamjotSingh"
+ms.prod: "microsoft-teams"
+doc_type: "apiPageType"
+---
+
+# Get chatMessage
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Retrieve the properties and relationships of chatmessage object.
```
Comment (Lauragra) [inline api-reference/beta/api/chatmessage-get.md]: Change "of chatmessage" to "of a chatMessage" and make the resource name a link. Pls. apply the same change to the description attribute in the YAML header.

File: api-reference/beta/api/chatmessage-list-hostedcontents.md [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+author: "RamjotSingh"
+ms.prod: "microsoft-teams"
+doc_type: "apiPageType"
+---
+
+# List hostedContents
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Retrieve a list of chatmessagehostedcontent objects.
```
Comment (Lauragra) [inline api-reference/beta/api/chatmessage-list-hostedcontents.md]: chatmessagehostedcontent -> chatMessageHostedContent, and add the link.

File: api-reference/beta/api/chatmessage-list-replies.md [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+author: "RamjotSingh"
+ms.prod: "microsoft-teams"
+doc_type: "apiPageType"
+---
+
+# List replies
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Retrieve a list of chatmessage objects.
```
Comment (Lauragra) [inline api-reference/beta/api/chatmessage-list-replies.md]: Please customize the summary for the API. Be sure to update the description as well.

File: api-reference/beta/api/chatmessage-post-replies.md [hunk 16]
```diff
@@ -0,0 +1,114 @@
...
+author: "RamjotSingh"
+ms.prod: "microsoft-teams"
+doc_type: "apiPageType"
+---
+
+# Create chatMessage
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Use this API to create a new chatMessage.
```
Comment (Lauragra) [inline api-reference/beta/api/chatmessage-post-replies.md]: Revise as "Create a new chatMessage object." Add the link to the resource.

File: api-reference/beta/api/chatmessage-post-replies.md [hunk 17]
```diff
@@ -0,0 +1,114 @@
...
+
+## Request headers
+
+| Name          | Description   |
+|:--------------|:--------------|
+| Authorization | Bearer {code} |
+
+## Request body
+
+In the request body, supply a JSON representation of [chatMessage](../resources/chatmessage.md) object.
```
Comment (Lauragra) [inline api-reference/beta/api/chatmessage-post-replies.md]: Add the word "a" after "of".

File: api-reference/beta/api/chatmessage-post-replies.md [hunk 18]
```diff
@@ -0,0 +1,114 @@
...
+
+## Response
+
+If successful, this method returns `201, Created` response code and a new [chatMessage](../resources/chatmessage.md) object in the response body.
```
Comment (Lauragra) [inline api-reference/beta/api/chatmessage-post-replies.md]: Add the word "a" after returns and remove the comma after 201.

File: api-reference/beta/api/chatmessagehostedcontent-get.md [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+author: "RamjotSingh"
+ms.prod: "microsoft-teams"
+doc_type: "apiPageType"
+---
+
+# Get chatMessageHostedContent
+
+[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]
+
+Retrieve the properties and relationships of chatmessagehostedcontent object.
```
Comment (Lauragra) [inline api-reference/beta/api/chatmessagehostedcontent-get.md]: Update the casing of the resource name and add the link.

File: api-reference/beta/api/channel-post-messages.md [hunk 20]
```diff
@@ -1,76 +1,86 @@
 ---
-title: "Send a message in a channel"
-description: "Send a new message in the specified channel."
-author: "nkramer"
+title: "Create chatMessage"
+description: "Use this API to create a new chatMessage."
```
Comment (Lauragra) [inline api-reference/beta/api/channel-post-messages.md]: Update the description here to match the summary.

File: api-reference/beta/resources/chatmessageimportance.md [hunk 21]
```diff
@@ -0,0 +1,21 @@
+---
+title: "chatMessageImportance resource type"
+description: "Represents the importance of an individual chat message within a channel or chat entity."
+localization_priority: Normal
+author: "RamjotSingh"
+ms.prod: "microsoft-teams"
+---
+
+# chatMessageImportance resource type
```
Comment (Lauragra) [inline api-reference/beta/resources/chatmessageimportance.md]: Is this an enumeration topic? Pls. see the guidance for documenting enumerations, here: https://msgo.azurewebsites.net/add/document/guidelines/enumerations.html.

File: api-reference/beta/resources/chatmessagetype.md [hunk 22]
```diff
@@ -0,0 +1,19 @@
+---
+title: "chatMessageType resource type"
+description: "Represents type of an individual chat message within a channel or chat entity."
+localization_priority: Normal
+author: "RamjotSingh"
+ms.prod: "microsoft-teams"
+---
+
+# chatMessageType resource type
```
Comment (Lauragra) [inline api-reference/beta/resources/chatmessagetype.md]: See the guidance for enumerations.

File: api-reference/beta/resources/team.md [hunk 23]
```diff
@@ -35,7 +35,8 @@ For more information about working with groups and members in teams, see [Use th
 |[Add app to team](../api/teamsappinstallation-add.md) | [teamsAppInstallation](teamsappinstallation.md) | Adds (installs) an app to a team.|
 |[Add tab to channel](../api/teamstab-add.md) | [teamsTab](../resources/teamstab.md) | Adds (installs) a tab to a team's channel.|
 |[List channel messages](../api/channel-list-messages.md)  | [chatMessage](../resources/chatmessage.md) | [Get messages in a channel](../api/channel-list-messages.md) |
-|[Send channel message](../api/channel-post-chatmessage.md)  | [chatMessage](../resources/chatmessage.md) | [Send a message to a channel](../api/channel-post-chatmessage.md) |
+|[Send channel message](../api/channel-post-messages.md)  | [chatMessage](../resources/chatmessage.md) | Send a message to a channel. |
+|[Send reply to a message in a channel](../api/channel-post-messagereply.md) | [chatMessage](../resources/chatmessage.md) | Reply to a message in a channel. |
```
Comment (Lauragra) [inline api-reference/beta/resources/team.md]: The name of this method should match the title - "Reply to a message in a channel".

File: concepts/changelog.md [hunk 24]
```diff
@@ -13,6 +13,14 @@ For details about known issues with Microsoft Graph APIs, see [Known issues](kno
 
 ## May 2019
 
+### Microsoft Teams APIs
```
Comment (Lauragra) [inline concepts/changelog.md]: Does this changelog entry need to move to June, or were the APIs made available in May?

File: api-reference/beta/resources/chatmessagetype.md (hunk 22 above)
Comment (RamjotSingh) [reply api-reference/beta/resources/chatmessagetype.md]: Removed the file and used the option 2 (inline the details).

File: api-reference/beta/resources/chatmessageimportance.md (hunk 21 above)
Comment (RamjotSingh) [reply api-reference/beta/resources/chatmessageimportance.md]: Removed the file and used the option 2 (inline the details).

File: concepts/changelog.md (hunk 24 above)
Comment (RamjotSingh) [reply concepts/changelog.md]: The APIs were shipped in May end.

File: api-reference/beta/api/channel-post-messages.md [hunk 25]
```diff
@@ -1,76 +1,86 @@
...
+title: "Create chatMessage"
+description: "Create a new chatMessage in the specified channel."
 localization_priority: Normal
+author: "RamjotSingh"
 ms.prod: "microsoft-teams"
+doc_type: "apiPageType"
 ---
 
-# Send a message to a channel
+# Send a message to a channel
```
Comment (Lauragra) [inline api-reference/beta/api/channel-post-messages.md]: The title here should be Create chatMessage.

File: api-reference/beta/api/chatmessage-list-hostedcontents.md [hunk 26]
```diff
@@ -0,0 +1,102 @@
+---
```
Comment (Lauragra) [inline api-reference/beta/api/chatmessage-list-hostedcontents.md]: Is this the file name that was generated by the tool, as opposed to chatmessagehostedcontents-list? I'm wondering whether the title should be "List chatMessageHostedContents".

Comment (RamjotSingh) [reply api-reference/beta/api/chatmessage-list-hostedcontents.md]: Yep all the added files were generated by the stub generator.

File: api-reference/beta/api/chatmessage-post-replies.md [hunk 27]
```diff
@@ -0,0 +1,114 @@
+---
+title: "Create chatMessage"
+description: "Creates a new chatMessage object in reply to an existing chatMessage object."
+localization_priority: Normal
+author: "RamjotSingh"
+ms.prod: "microsoft-teams"
+doc_type: "apiPageType"
+---
+
+# Create chatMessage
```
Comment (Lauragra) [inline api-reference/beta/api/chatmessage-post-replies.md]: This is another duplicate title. Could be "Create chatMessage reply"?

File: api-reference/beta/api/chatmessage-post-replies.md [hunk 28]
```diff
@@ -0,0 +1,114 @@
...
+### Request
+
+The following is an example of the request.
+<!-- {
+  "blockType": "request",
+  "name": "create_chatmessage_from_chatmessage"
+}-->
+
+```http
+POST https://graph.microsoft.com/beta/teams/{id}/channels/{id}/messages
```
Comment (Lauragra) [inline api-reference/beta/api/chatmessage-post-replies.md]: The example doesn't match the syntax in the HTTP request section.

File: api-reference/beta/resources/channel.md [hunk 29]
```diff
@@ -25,7 +25,8 @@ where files are shared, and where tabs are added.
 |[Update channel](../api/channel-patch.md) | [channel](channel.md) | Update properties of the channel.|
 |[Delete channel](../api/channel-delete.md) | None | Delete a channel.|
 |[List channel messages](../api/channel-list-messages.md)  | [chatMessage](../resources/chatmessage.md) | Get messages in a channel |
-|[Send channel message](../api/channel-post-chatmessage.md)  | [chatMessage](../resources/chatmessage.md) | [Send a message to a channel](../api/channel-post-chatmessage.md) |
+|[Send channel message](../api/channel-post-messages.md) | [chatMessage](../resources/chatmessage.md) | Send a message to a channel. |
```
Comment (Lauragra) [inline api-reference/beta/resources/channel.md]: Please update method titles to match API topic titles.

File: api-reference/beta/api/chatmessage-post-replies.md (hunk 28 above)
Comment (RamjotSingh) [reply api-reference/beta/api/chatmessage-post-replies.md]: Fixed.

Comment (RamjotSingh) [reply api-reference/beta/api/chatmessage-post-replies.md]: Fixed.

File: api-reference/beta/resources/chatmessage.md [hunk 30]
```diff
@@ -21,20 +21,20 @@ The message can be a root message or part of a thread that is defined by the **r
 |[Get Channel message](../api/channel-get-message.md) | [chatmessage](chatmessage.md) | Get a single root message from a channel.|
 |[List replies to a message](../api/channel-list-messagereplies.md) | [chatmessage](chatmessage.md) collection| Get the list of all replies to a message in channel.|
 |[Get a reply to a message](../api/channel-get-messagereply.md) | [chatmessage](chatmessage.md)| Get a single reply to a message in a channel.|
-|[Send a message in a channel](../api/channel-post-chatmessage.md) | [chatmessage](chatmessage.md)| Create a new top-level message in a channel.|
-|[Reply to a message in a channel](../api/channel-post-messagereply.md) | [chatmessage](chatmessage.md)| Reply to an existing message in a channel.|
+|[Send channel message](../api/channel-post-messages.md) | [chatmessage](chatmessage.md)| Create a new top-level message in a channel.|
```
Comment (Lauragra) [inline api-reference/beta/resources/chatmessage.md]: Update method titles.

Comment (RamjotSingh) [reply api-reference/beta/resources/chatmessage.md]: Done here, in channel.md and team.md.

**Comment by RamjotSingh** [comment]
@Lauragra Do you have any more comments? Current one are all resolved.

**Comment by Lauragra** [comment]
@RamjotSingh , no, but we need to resolve the validation error before we can merge/publish. Can you take a look please?

**Pull request merged**